### PR TITLE
Fix issues uncovered by new linting rules introduced in golangci-lint

### PR DESCRIPTION
**Pull request by msnook**
### Fixes Linting Issues

[New rules](https://github.com/golangci/golangci-lint/commit/bf27481efd4b1916eac1fedac09077dd82dd995f) were recently added to `golangci-lint`.

### Change Details

- Updated code to adhere to `S1039` and `SA5011` static analysis rules.
- Disabled linting rule `SA1029` due to incompatibilities wit SSAS logging.

### Security Implications

No PHI/PII is affected by this change.  This is a code improvement task to adhere to latest static analysis rules.

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

### Acceptance Validation

Local validation included `make lint` and `make unit-test`.

### Feedback Requested

Please review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ package:
 	-v ${PWD}:/go/src/github.com/CMSgov/bcda-ssas-app packaging $(version)
 
 lint:
-	docker-compose -f docker-compose.test.yml run --rm tests golangci-lint --deadline=3m run ./...
+	docker-compose -f docker-compose.test.yml run --rm tests golangci-lint --deadline=3m -e SA1029 run ./...
```
Comment (dhgreene) [inline Makefile]: I'm curious why this is disabled in this repo and not in the[ associated ACO API PR](https://github.com/CMSgov/bcda-app/pull/465).

Comment (msnook) [reply Makefile]: The technique that was performed in the ACO API PR was reproduced in SSAS, and works fine... however, the is some logging in SSAS which does not play well with ContextKey.  I could not get it to work.  Maybe we should write a tech debt ticket to address it later?

Comment (msnook) [reply Makefile]: Reference: https://travis-ci.org/github/CMSgov/bcda-ssas-app/builds/666583174

**Comment by msnook** [comment]
Tech debt ticket: https://jira.cms.gov/browse/BCDA-2876

**Pull request merged**